File: C#_Base_PT4/XString58.cs
// File: "XString58"
using PT4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XString58");

            string s = GetString();

            int n2 = s.LastIndexOf('.');

            int n1 = s.LastIndexOf('\\');
            string sstr;
            if(n2 != -1) 
                sstr = s.Substring(n1+1, n2-n1-1);
            else 
                sstr = s.Substring(n1+1);
            Put(sstr);
        }
    }
}

File: C++_Base_PT4/XString58.cpp
#include "pt4.h"
using namespace std;

void Solve()
{
    Task("XString58");
    
    string path; 
    
    pt >> path;
	
	int l = path.length();
  
  	int i;
  
    int k = l - 1;
    
  	for(i = l - 1; i >=0; i--)
	{
		if(path[i] == '.') k = i;
   		if(path[i] == '\\') break;
    }
    
    pt << path.substr(i+1, k-i-1);

}
